lib/model/store.dart

```diff
@@ -570,7 +572,18 @@ class LiveGlobalStore extends GlobalStore {
 
   @override
   Future<Account> doInsertAccount(AccountsCompanion data) async {
-    final accountId = await _db.createAccount(data); // TODO(log): db errors
+    final int accountId;
+    try {
+      accountId = await _db.createAccount(data); // TODO(log): db errors
+    } catch (e) {
+      if (e case DriftRemoteException(
+            remoteCause: SqliteException(
+              extendedResultCode: SqlExtendedError.SQLITE_CONSTRAINT_UNIQUE))) {
+          throw AccountAlreadyExistsException();
```

Comment:
Rereading this, there are enough details here that I'm realizing we should definitely have a test for it. Or I guess two tests: one for duplicate realmUrl and email, and one for duplicate realmUrl and userId.

That also causes me to notice that the best home for this logic would be in our `database.dart`, inside this `AppDatabase.createAccount` method. That way these rather database-specific details go in our database layer.

So then the tests can be: create one account, create another that should collide with it, check that the second one throws an AccountAlreadyExistsException.